ASC-157 Add dead tap test

johnduarte opened the pull request:
This commit adds a test to ensure that no "dead taps" are present
on the deployment host.

phongdly commented on molecule/default/tests/test_bridges.py:
```diff
@@ -0,0 +1,22 @@
+def test_for_dead_taps(host):
+    """Ensure OpenVSwitch bridges have at least 2 open ports"""
+
+    cmd = 'ovs-vsctl list-br'
```
I am curious how `ovs-vsctl` was setup, tried the command on `infra1` on phobos MNAIO and it doesn't seem to be able to run the cli:
I am curious how `ovs-vsctl` was setup, tried the command on `infra1` on phobos MNAIO and it doesn't seem to be able to run the cli:

```
root@infra1:~# ovs-vsctl list-br
-bash: ovs-vsctl: command not found
 root@infra1:~#
```

johnduarte replied:
`ovs-vsctl` is installed as part of the `openvswitch-switch` package in the `main/tasks.yml` that is part of this PR.

phongdly commented on molecule/default/tests/test_dead_taps.py:
```diff
@@ -0,0 +1,18 @@
+    """Ensure there are no dead tap interfaces in OpenVSwitch"""
+
+    cmd = 'ovs-vsctl show'
+    res = host.run(cmd)
```
I am curious how `ovs-vsctl` was setup, tried the command on `infra1` on phobos MNAIO and it doesn't seem to be able to run the cli:
I am curious how `ovs-vsctl` was setup, tried the command on `infra1` on phobos MNAIO and it doesn't seem to be able to run the cli:

```
root@infra1:~# ovs-vsctl show
-bash: ovs-vsctl: command not found
 root@infra1:~#
```

johnduarte commented:
@phongdly does my previous response satisfy your curiosity?

phongdly commented:
@johnduarte yes, sorry I missed it in the first review.
 
Ran the PR on my mnaio env and see some errors here: https://gist.github.com/phongdly/383e5eed7c973632796d0197ac2cb017
Also, please fix the conflict file.

johnduarte commented:
@phongdly I have corrected the conflict and addressed the failure in the bridge test. In my MNAIO environment, I was able to duplicate the failure you saw. It is now passing for me. Thanks.

phongdly commented:
Passed local test on Phobos MNAIO:
    ============================= test session starts ==============================
    platform linux2 -- Python 2.7.12, pytest-3.5.0, py-1.5.3, pluggy-0.6.0
    rootdir: /root/github/rpc-openstack-system-tests/molecules/molecule-rpc-openstack-post-deploy/molecule/default, inifile:
    plugins: testinfra-1.7.1, rpc-0.5.0
collected 37 items

    tests/test_dhcp_agents.py .                                              [  2%]
    tests/test_for_acs-295.py ssssss                                         [ 18%]
    tests/test_hypervisor_free.py ..                                         [ 24%]
    tests/test_mysql_replication.py s                                        [ 27%]
    tests/test_bridges.py .                                                  [ 29%]
    tests/test_dead_taps.py .                                                [ 32%]
    tests/test_env_scan.py ......                                            [ 48%]
    tests/test_for_acs-150.py sssssssssssssssssss                            [100%]

     generated xml file: /root/github/rpc-openstack-system-tests/molecules/molecule-rpc-openstack-post-deploy/molecule/default/test.xml
    ==================== 11 passed, 26 skipped in 32.50 seconds ====================
Verifier completed successfully.
